render_silhouette_regmarks.py

```diff
@@ -0,0 +1,142 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA.
+#
+"""
+Base module for rendering regmarks for Silhouette CAMEO products in Inkscape.
+"""
+
+import sys, os
+# Enables stand alone mode and helps for tests #
+# We append the directory where this script lives and inkscape extension folder to sys.path
+sys.path.append(os.path.dirname(os.path.abspath(sys.argv[0])))
+sys_platform = sys.platform.lower()
+if sys_platform.startswith("win"):
+    sys.path.append(r"C:\Program Files\Inkscape\share\inkscape\extensions")
+elif sys_platform.startswith("darwin"):
+    sys.path.append("/Applications/Inkscape.app/Contents/Resources/share/inkscape/extensions")
+else:   # linux
+    sys.path.append("/usr/share/inkscape/extensions")
+
+import inkex
+from inkex import EffectExtension, Boolean, Rectangle, PathElement, Layer, Group, TextElement, Transform
+from gettext import gettext
+from inkex.units import convert_unit
+
+REGMARK_LAYERNAME = 'Regmarks'
+REGMARK_LAYER_ID = 'regmark'
+REGMARK_TOP_LEFT_ID = 'regmark-tl'
+REGMARK_TOP_RIGHT_ID = 'regmark-tr'
+REGMARK_BOTTOM_LEFT_ID = 'regmark-bl'
+REGMARK_SAFE_AREA_ID = 'regmark-safe-area'
+REGMARK_NOTES_ID = 'regmark-notes'
+
+REG_SQUARE_MM = 5
+REG_LINE_MM = 20
+REG_SAFE_AREA_MM = 20
+
+# https://www.reddit.com/r/silhouettecutters/comments/wcdnzy/the_key_to_print_and_cut_success_an_extensive/
+# > The registration mark thickness is actually very important. For some reason, 0.3 mm marks work perfectly. 
+# > The thicker you get, the less accurate registration will be. ~~~ galaxyman47
+REG_MARK_LINE_WIDTH_MM = 0.3
+
+REG_MARK_INFO_FONT_SIZE_PX = 2.5
+
+ENABLE_CHECKERBOARD = True
+
+class InsertRegmark(EffectExtension):
+	def add_arguments(self, pars):
+		# Parse arguments
+		pars.add_argument("-X", "--reg-x", "--regwidth",  type = float, dest = "regwidth",   default = 0.0, help="X mark to mark distance [mm]")
+		pars.add_argument("-Y", "--reg-y", "--reglength", type = float, dest = "reglength",  default = 0.0, help="Y mark to mark distance [mm]")
+		pars.add_argument("--rego-x",  "--regoriginx",    type = float, dest = "regoriginx", default = 10.0,  help="X mark origin from left [mm]")
+		pars.add_argument("--rego-y", "--regoriginy",     type = float, dest = "regoriginy", default = 10.0,  help="X mark origin from top [mm]")
+		pars.add_argument("--verbose", dest = "verbose",  type = Boolean, default = False, help="enable log messages")
+
+	def effect(self):
+		reg_origin_X = self.options.regoriginx
+		reg_origin_Y = self.options.regoriginy
+		reg_width = self.options.regwidth if self.options.regwidth else self.svg.to_dimensional(self.svg.viewport_width, "mm") - reg_origin_X * 2
+		reg_length = self.options.reglength if self.options.reglength else self.svg.to_dimensional(self.svg.viewport_height, "mm") - reg_origin_Y * 2
```

Comment:
can you switch from `... if ... else ...` to `... or ...`? this should work for 0 :)